File: prime.java
import java.util.ArrayList;
import java.util.List;

public class prime {
    
    

    public static void main(String[] args) {
        System.out.print(largest_prime(1000000));
        
    }

    public static List<Integer> factors(int n) {
        List<Integer> factors = new ArrayList<>();
        for (int i = 1; i <= n; i++){
            if (n % i == 0){
                factors.add(i);
            }
        }
        return factors;
    }

    public static boolean isPrime(int n) {
        
        if (factors(n).size()==2){
            return true;
        } else{
            return false;
        }

    }

    public static Integer largest_prime(int n){
        
        int largest_prime = 2;
        for (int i = 999000; i<=n; i++){
            if (isPrime(i) == true){
                largest_prime = i;
            } 
        }
        return largest_prime;
        
    

    }


}

File: prime.kt
object prime {
    @JvmStatic
    fun main(args: Array<String>) {
        println(largest_prime(300000))
    }

    fun factors(n: Int): List<Int> {
        val factors: MutableList<Int> = java.util.ArrayList<Int>()
        for (i in 1..n) {
            if (n % i == 0) {
                factors.add(i)
            }
        }
        return factors
    }

    fun isPrime(n: Int): Boolean {
        return if (factors(n).size == 2) {
            true
        } else {
            false
        }
    }

    fun largest_prime(n: Int): Int {
        var largest_prime = 2
        for (i in 199999..n) {
            if (isPrime(i) == true) {
                largest_prime = i
            }
        }
        return largest_prime
    }
}